src/components/structures/RoomView.js

```diff
@@ -77,13 +77,6 @@ module.exports = React.createClass({
     },
 
     componentWillUnmount: function() {
-        if (this.refs.messagePanel) {
-            var messagePanel = ReactDOM.findDOMNode(this.refs.messagePanel);
-            messagePanel.removeEventListener('drop', this.onDrop);
-            messagePanel.removeEventListener('dragover', this.onDragOver);
-            messagePanel.removeEventListener('dragleave', this.onDragLeaveOrEnd);
-            messagePanel.removeEventListener('dragend', this.onDragLeaveOrEnd);
-        }
```

Comment:
Whilst I agree that componentWillUnmount means that the underlying DOM Node is about to be removed from the DOM and so the event listeners will be cleaned up, it feels *really* wrong to me to skip the cleanup here, even though it's not going to achieve much.  Just as you'd always free() stuff before calling exit() at the end of a program, I would want to remove the event listeners - just in case the DOM element does somehow keep hanging around as an orphan and keeping a reference to the listener alive (or vice versa) or whatever.  More importantly, FB's own examples for best practices for managing DOM event listeners show them being explicitly removed: https://facebook.github.io/react/tips/dom-event-listeners.html.  So I would certainly consider keeping them here, perhaps with a comment that they are purely hygienic, but to make it clear they have been consciously considered rather than forgotten.